languages/elixir/exercises/concept/agents/.docs/introduction.md

```diff
@@ -0,0 +1,23 @@
+The `Agent` module facilitates an abstraction for spawning processes and the _receive_, _send_ loop -- from here, we will call processes started using the `Agent` module _'agent processes'_. An _agent process_ might be chosen to represent a central shared state.
+
+To start an unlinked _agent process_, `Agent` provides the `start/2` function:
+
+```elixir
+# Start an agent process with an initial value of an empty list
+{:ok, agent_pid} = Agent.start(fn -> [] end)
+```
+
+Just like `Map` or `List`, `Agent` provides many functions for working with _agent processes_:
+
+```elixir
+# Get the current state of the agent process started previously
+Agent.get(agent_pid, fn list -> list end)
+# => []
+
+# Add an element to the list
+item = :element
+Agent.update(agent_pid, fn list -> [item | list] end)
+# => :ok
```

Comment:
I'm surprised that @ErikSchierboom didn't complain that none of this is new syntax. Maybe we could keep just the single example with `Agent.start`?

Reply:
I'm trying to apply the optimistic merging strategy more. But you are right of course :)

Reply:
I can't speak for Tim, but I would prefer to know up front that something in my PR needs fixing. Optimistic merging is supposed to mean that we merge "good enough", and fix problems later, but how can we fix the problems if we aren't aware of them and don't track them with issues?

Reply:
Isn't it new syntax? Where have we shown Agent.get and Agent.update before?

Reply:
We showed already how to call a function on a module and how to pass arguments to that function, so... no new syntax. New function is not new syntax.

Reply:
Precisely. What we usually do in these cases is to explicitly mention the module/class that contains relevant functionality, in this case `Agent`.

Reply:
So, just as a thought experiment, we showed that in `basics`, I would imagine there is a lot of revisions required throughout the track, no?

Reply:
> I would imagine there is a lot of revisions required throughout the track, no?

Well it might be less than you think. In many cases, you were introducing new syntax:

- https://github.com/exercism/v3/blob/master/languages/elixir/exercises/concept/strings/.docs/introduction.md
- https://github.com/exercism/v3/blob/master/languages/elixir/exercises/concept/numbers/.docs/introduction.md

https://github.com/exercism/v3/blob/master/languages/elixir/exercises/concept/enum/.docs/introduction.md is an example where no new syntax was introduced. I argued against including an example, but I think iHid was okay with it. @angelikatyborska do you remember what was said there?

Reply:
Jeremy gave a good explanation as to why that rule was put in place (https://github.com/exercism/v3/pull/1587#discussion_r437363290) and it convinced me to delete all the code snippets from that introduction but one (https://github.com/exercism/v3/pull/1587#discussion_r437382255)

Reply:
Ah yes, that was it. Thanks! @neenjaw does the comment by Jeremy help explain why we have this rule?